runtime/src/lib.rs

```diff
@@ -1,209 +1,10 @@
-    // ~50 times faster than using HashSet for very short slices.
-    for i in 1..xs.len() {
-        if xs[i..].contains(&xs[i - 1]) {
-            return true;
-        }
-    }
-    false
-}
-
-/// Get mut references to a subset of elements.
-fn get_subset_unchecked_mut<'a, T>(xs: &'a mut [T], indexes: &[u8]) -> Vec<&'a mut T> {
-    // Since the compiler doesn't know the indexes are unique, dereferencing
-    // multiple mut elements is assumed to be unsafe. If, however, all
-    // indexes are unique, it's perfectly safe. The returned elements will share
-    // the liftime of the input slice.
-
-    // Make certain there are no duplicate indexes. If there are, panic because we
-    // can't return multiple mut references to the same element.
-    if has_duplicates(indexes) {
-        panic!("duplicate indexes");
-    }
-
-    indexes
-        .iter()
-        .map(|i| {
-            let ptr = &mut xs[*i as usize] as *mut T;
-            unsafe { &mut *ptr }
-        })
-        .collect()
-}
-
-/// Execute a transaction.
-/// This method calls each instruction in the transaction over the set of loaded Accounts
-/// The accounts are committed back to the bank only if every instruction succeeds
-pub fn execute_transaction(
-    tx: &Transaction,
-    loaders: &mut [Vec<(Pubkey, Account)>],
-    tx_accounts: &mut [Account],
-    tick_height: u64,
-) -> Result<(), RuntimeError> {
-    for (instruction_index, instruction) in tx.instructions.iter().enumerate() {
-        let executable_accounts = &mut (&mut loaders[instruction.program_ids_index as usize]);
-        let mut program_accounts = get_subset_unchecked_mut(tx_accounts, &instruction.accounts);
-        execute_instruction(
-            tx,
-            instruction_index,
-            executable_accounts,
-            &mut program_accounts,
-            tick_height,
-        )
-        .map_err(|err| RuntimeError::ProgramError(instruction_index as u8, err))?;
-    }
-    Ok(())
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn test_has_duplicates() {
-        assert!(!has_duplicates(&[1, 2]));
-        assert!(has_duplicates(&[1, 2, 1]));
-    }
-
-    #[test]
-    fn test_get_subset_unchecked_mut() {
-        assert_eq!(get_subset_unchecked_mut(&mut [7, 8], &[0]), vec![&mut 7]);
-        assert_eq!(
-            get_subset_unchecked_mut(&mut [7, 8], &[0, 1]),
-            vec![&mut 7, &mut 8]
-        );
-    }
-
-    #[test]
-    #[should_panic]
-    fn test_get_subset_unchecked_mut_duplicate_index() {
-        // This panics, because it assumes duplicate detection is done elsewhere.
-        get_subset_unchecked_mut(&mut [7, 8], &[0, 0]);
-    }
-
-    #[test]
-    #[should_panic]
-    fn test_get_subset_unchecked_mut_out_of_bounds() {
-        // This panics, because it assumes bounds validation is done elsewhere.
-        get_subset_unchecked_mut(&mut [7, 8], &[2]);
-    }
-}
+mod accounts;
+pub mod bank;
+pub mod bloom;
+pub mod genesis_block;
```

Comment:
@mvines, want this in `solana_runtime` or `solana_sdk`?  I'm thinking SDK might be better.

Reply:
☑️ sdk